docs/utilities/idempotency.md

```diff
@@ -0,0 +1,952 @@
+
+```java hl_lines="2-8" title="Customizing IdempotencyConfig"
+IdempotencyConfig.builder()
+                .withEventKeyJMESPath("id")
+                .withPayloadValidationJMESPath("paymentId")
+                .withThrowOnNoIdempotencyKey(true)
+                .withExpirationInSeconds(5 * 60) // 5 minutes
+                .withUseLocalCache(false)
+                .withLocalCacheMaxItems(432)
+                .withHashFunction("SHA-256")
+                .build()
+```
+
+These are the available options for further configuration:
+
+| Parameter                                         | Default | Description                                                                                                                      |
+|---------------------------------------------------|---------|----------------------------------------------------------------------------------------------------------------------------------|
+| **EventKeyJMESPath**                              | `""`    | JMESPath expression to extract the idempotency key from the event record. See available [built-in functions](utilities) |
+| **PayloadValidationJMESPath**                     | `""`    | JMESPath expression to validate whether certain parameters have changed in the event                                             |
+| **ThrowOnNoIdempotencyKey**                       | `False` | Throw exception if no idempotency key was found in the request                                                                   |
+| **ExpirationInSeconds**                           | 3600    | The number of seconds to wait before a record is expired                                                                         |
+| **UseLocalCache**                                 | `true`  | Whether to locally cache idempotency results (LRU cache)                                                                         |
+| **LocalCacheMaxItems**                            | 256     | Max number of items to store in local cache                                                                                      |
+| **HashFunction**                                  | `MD5`   | Algorithm to use for calculating hashes, as supported by `java.security.MessageDigest` (eg. SHA-1, SHA-256, ...)                 |
+
+These features are detailed below.
+
+### Handling concurrent executions with the same payload
+
+This utility will throw an **`IdempotencyAlreadyInProgressException`** if we receive **multiple invocations with the same payload while the first invocation hasn't completed yet**.
+
+!!! info
+    If you receive `IdempotencyAlreadyInProgressException`, you can safely retry the operation.
+
+This is a locking mechanism for correctness. Since we don't know the result from the first invocation yet, we can't safely allow another concurrent execution.
+
+### Using in-memory cache
+
+**By default, in-memory local caching is enabled**, to improve performance of your Lambda function. 
```

Comment:
Following your suggestion, it is now enabled by default: https://github.com/jeromevdl/aws-lambda-powertools-java/blob/4855a2089a8750406343c6deb944eecb88e815a8/powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/IdempotencyConfig.java#L83

Reply:
I now understand why it is disable by default. "Its disabled by default to avoid using memory in an unpredictable way."

We need to change the docs to reflect that.

Reply:
reverting to disabled by default